### PR TITLE
Proposed edits for #120

### PR DESCRIPTION
**Pull request by JustinCappos**
This is the proposed set of edits for #120.  Further refinement is most welcome.

### CODE REVIEW

**Comment by jhdalek55** [comment]
I can not pass judgement on the technical merits of this PR, but I did notice a small typo in line 497. Fix the spelling of Uptane (p not o)

Also, since I will be capitalizing for consistency shortly, please capitalize the "D" in Director keys---as I'm assuming these are the keys from the Director repository.

**Comment by mnm678** [comment]
I have a couple of concerns:

First, I worry that this creates an opportunity to replay director metadata, especially in the first situation. Replaying a location based update from one region in another region could cause problems.

Second, how would a vehicle know when to install a location based update if it is not specified by the director? I might be missing something, but if the director is sending the metadata after receiving a vehicle version manifest, the director knows the ECU id for the vehicle and should be able to include this in the metadata.

**Comment by iramcdonald** [comment]
Hi Marina,

Good comments.

The IEEE 1609 use case is that the Director pushes the *non-firmware*
content to a CDN server geographically near a location (in cellular network
terms, in the "fog").  And the vehicles (by some magic?) query that CDN
server and download the content/metadata without usual one-at-a-time
exchanges w/ the Director.  Such content as current traffic conditions
would never be archived on the OEM's Image Repository (because
they're ephemeral safety content).

Although Justin has proposed some good text, I'm actually opposed to
this change before Uptane Standard v1.0, based on the criticisms that
Jon Oster and you have advanced.  Nice capability, but pretty radically
different from the current Uptane model.

Cheers,
- Ira

Ira McDonald (Musician / Software Architect)
Co-Chair - TCG Trusted Mobility Solutions WG
Co-Chair - TCG Metadata Access Protocol SG
Chair - Linux Foundation Open Printing WG
Secretary - IEEE-ISTO Printer Working Group
Co-Chair - IEEE-ISTO PWG Internet Printing Protocol WG
IETF Designated Expert - IPP & Printer MIB
Blue Roof Music / High North Inc
http://sites.google.com/site/blueroofmusic
http://sites.google.com/site/highnorthinc
mailto: blueroofmusic@gmail.com
PO Box 221  Grand Marais, MI 49839  906-494-2434

**Comment by jhdalek55** [comment]
We need a definitive decision on whether this commit should be flagged for 1.0 or held off. There are mixed opinions here.

Lois

**Comment by tkfu** [comment]
Strongly opposed to including it in 1.0. It goes beyond just figuring out the places where ECU IDs are mentioned; because it's a major change we really need to give it time for people to chew on, think about implementation details, and generally feel out the idea until the corner cases we haven't thought of yet have a chance to present themselves. 

For example, how should version numbers be dealt with in the location-focused updates use case? I think that Justin's suggestion (in #120) of this being a good use case for a 3rd repository was at least partially made in anticipation of the difficulties with versioning, but I'd want to see that fleshed out a bit more. What would the repository mapping metadata look like for that use case?

Given that Ira said the image repo wouldn't enter into it at all for this use case, and that the director repo also likely wouldn't be involved (because of versioning difficulties), does it even really make sense for it to be part of Uptane? It's then effectively just a TUF repository with an online Targets role, because it would be completely independent of the firmware updating infrastructure, and it wouldn't target specific vehicles.

Basically, I think there's a lot more to this than meets the eye, and it's far too much to consider including at the last minute.

**Comment by iramcdonald** [comment]
Hi,

Once again, I agree with all the points Jon made and and opposed to
including this
in Uptane v1.0.

About the Image Repository's function here:  At least in the case of local
traffic updates
or local regulatory info, clearly the OEM's not an appropriate
authoritative source - the
ITS infrastructure (outside the OEM's domain) is the authoritative source.
How that
would affect Director interactions and how the versioning would be resolved
are not
simple problems.

Cheers,
- Ira

Ira McDonald (Musician / Software Architect)
Co-Chair - TCG Trusted Mobility Solutions WG
Co-Chair - TCG Metadata Access Protocol SG
Chair - Linux Foundation Open Printing WG
Secretary - IEEE-ISTO Printer Working Group
Co-Chair - IEEE-ISTO PWG Internet Printing Protocol WG
IETF Designated Expert - IPP & Printer MIB
Blue Roof Music / High North Inc
http://sites.google.com/site/blueroofmusic
http://sites.google.com/site/highnorthinc
mailto: blueroofmusic@gmail.com
PO Box 221  Grand Marais, MI 49839  906-494-2434

**Comment by JustinCappos** [comment]
I don't mind waiting for 1.1.  I certainly wouldn't want us to merge this (or other PRs) with any strong objections.

**Comment by jhdalek55** [comment]
@JustinCappos  Since we have reframed Issue #120 to be concerned only with issues of the factory flashing/updating use case for non-ECU-targeted updates should this PR perhaps be closed?
